File: PickerView/app/src/main/java/com/example/pickerview/MainActivity.java
package com.example.pickerview;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.NumberPicker;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        NumberPicker numberPicker = findViewById(R.id.numberPicker);
        NumberPicker numberPicker2 = findViewById(R.id.numberPicker2);
        final TextView textView = findViewById(R.id.textView4);
        final Person person = new Person(170, 70);

        textView.setText("Value:\n" + person.calculateBmi());
        numberPicker.setMinValue(30);
        numberPicker.setMaxValue(130);
        numberPicker2.setMinValue(50);
        numberPicker2.setMaxValue(210);
        numberPicker.setValue(70);
        numberPicker2.setValue(170);


        numberPicker.setOnValueChangedListener(new NumberPicker.OnValueChangeListener() {
            @Override
            public void onValueChange(NumberPicker picker, int oldVal, int newVal) {
                person.setWeight(newVal);
                textView.setText("Value:\n" + person.calculateBmi());
            }
        });

        numberPicker2.setOnValueChangedListener(new NumberPicker.OnValueChangeListener() {
            @Override
            public void onValueChange(NumberPicker picker, int oldVal, int newVal) {
                person.setHeight(newVal);
                textView.setText("Value:\n" + person.calculateBmi());
            }
        });
    }

}

File: BMIKotlin/app/src/main/java/com/example/bmikotlin/MainActivity.kt
package com.example.bmikotlin

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.NumberPicker
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val initCalc: InitializeCalculator
        val person = Person(150,70)
        initCalc = InitializeCalculator(textView4, numberPicker,numberPicker2,person)
        initCalc.numberp.setOnValueChangedListener({ picker, oldVal, newVal -> initCalc.calculate(newVal, true) })
        initCalc.numberp2.setOnValueChangedListener({ picker, oldVal, newVal -> initCalc.calculate(newVal, false) })

    }
}
